File: javascript/collatz-conjecture/collatz-conjecture.js
export const steps = (n) => {  
  if (n<1) throw Error("Only positive numbers are allowed")
  let counter = 0
  while (n > 1) {
    n = n % 2 === 0 ? n / 2 : n * 3 + 1
    counter++
  }
  return counter
};

File: go/collatz-conjecture/collatz_conjecture.go
package collatzconjecture

import "fmt"

func CollatzConjecture(n int) (int, error) {
	steps:= 0
	if n < 1 {
		return steps, fmt.Errorf("n must be positive")
	}
	for n > 1 {
		if (n%2 == 0) {
			n = n/2
		} else {
			n = n * 3 + 1
		}
		steps++
		}

	return steps, nil
}
